spec/user_index_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature do
+  let!(:users) do
+    [
+      User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10),
+      User.create(name: 'Ponobiom', photo: '/path/to/photo2.jpg', post_counter: 5),
+      User.create(name: 'Stonebwoy', photo: '/path/to/photo3.jpg', post_counter: 3)
+    ]
+  end
+
+  before do
+    visit root_path
+  end
+
+  it 'displays the username of all users' do
+    users.each do |user|
+      expect(page).to have_content(user.name)
+      expect(page).to have_css("img[src*='#{user.photo}']")
+      expect(page).to have_content(user.posts.count)
+      expect(page).to have_link(href: "users/#{user.id}")
+    end
+  end
```

Comment:
- [ ] **[OPTIONAL]** Kindly separate these assertions to be more organized. Assertions within an it-block should assert for related tests, but here you are testing for username, photo, posts counter and so on. Kindly divide these into their own it blocks for better organization.

- [ ] For the redirection test, kindly ensure you actually click the link and that it takes you to the page, I can see you are only checking for a link, and nothing more.

Reply:
@julie-ify done with this change